wails-frontend/src/pages/passwordPrompt.tsx

```diff
@@ -66,52 +69,64 @@ const PasswordPrompt = () => {
       }
       return ApplyPassword(password);
     }
-  };
+  }
 
-  const updatePassword = (e: React.ChangeEvent<HTMLInputElement>) =>
+  function updatePassword(e: React.ChangeEvent<HTMLInputElement>) {
     setPassword(e.target.value);
-  const updatePasswordConfirm = (e: React.ChangeEvent<HTMLInputElement>) =>
+  }
+  function updatePasswordConfirm(e: React.ChangeEvent<HTMLInputElement>) {
     setPasswordConf(e.target.value);
+  }
 
   return (
-    <section className="PasswordPrompt">
-      <div>{req.Msg}</div>
-      <div className="baseline">{baselineStr(req)}</div>
-      <div id="input" className="input-box">
-        <input
-          id="name"
-          className="input"
-          onChange={updatePassword}
-          autoComplete="off"
-          name="input"
-          type="password"
-          placeholder="Enter your password"
-        />
-      </div>
-      {(req.Action === promptAction.newPasswordReq ||
-        req.Action === promptAction.importReq) && (
-        <div id="input" className="input-box">
-          <input
-            id="name"
-            className="input"
-            onChange={updatePasswordConfirm}
-            autoComplete="off"
+    <div className="bg-primary min-h-screen">
+      <div
+        className="flex flex-col justify-center h-screen
+        max-w-xs min-w-fit text-f-primary m-auto"
+      >
+        <h1 className="mas-title">{req.Msg}</h1>
+        <p className="mas-body pt-4">{baselineStr(req)}</p>
+        <div className="pt-4">
+          <Password
+            onChange={updatePassword}
+            value={password}
             name="input"
-            type="password"
-            placeholder="Confirm your password"
+            placeholder="Enter your password"
           />
         </div>
-      )}
-      <div>
-        <button className="btn" onClick={handleCancel}>
-          Cancel
-        </button>
-        <button className="btn" onClick={applyPassword}>
-          {applyStr(req)}
-        </button>
+        {(req.Action === promptAction.newPasswordReq ||
+          req.Action === promptAction.importReq) && (
+          <div className="pt-4">
+            <Password
+              onChange={updatePasswordConfirm}
+              value={passwordConfirm}
+              name="input"
+              placeholder="Confirm your password"
+            />
+          </div>
+        )}
+        <p className="pt-4 mas-body">{resultMsg}</p>
+        <div className="pt-4 flex gap-4">
+          <div className="max-w-min">
+            <Button variant={'secondary'} onClick={handleCancel}>
+              Cancel
+            </Button>
+          </div>
+          <Button
+            preIcon={<FiLock />}
+            onClick={applyPassword}
+            disabled={
+              req.Action ===
+              (promptAction.newPasswordReq || promptAction.importReq)
+                ? !(password && passwordConfirm)
+                : !password
```

Comment:
Are you sure its working ??? 
imho
```
req.Action ===
    (promptAction.newPasswordReq || promptAction.importReq)
 ```
will always return false
maybe you can create a new function to factorize this:

const needConfirm = (req: promptRequest) => 
req.Action === promptAction.newPasswordReq  || req.Action === promptAction.importReq

Reply:
@peterjah I tested it in front and it works!

![image](https://github.com/massalabs/thyra-plugin-wallet/assets/72019005/2d7de250-4d00-49a0-b582-bdcf4eab1d2b)

![image](https://github.com/massalabs/thyra-plugin-wallet/assets/72019005/32dc413c-ed7b-4f83-a795-6010c4d1ed72)

![image](https://github.com/massalabs/thyra-plugin-wallet/assets/72019005/abda4d74-201b-46ee-b401-88f0f0e5af7f)

Reply:
in Logic :
`req.Action === (promptAction.newPasswordReq || promptAction.importReq)`

AND

``(req.Action === promptAction.newPasswordReq) || (req.Action === promptAction.importReq)

Reply:
Even this is working is a nightmare to understand. Let's extract that logic and make things easy to understand... we are mixing OR and ! just hell to follow...